packages/main/test/samples/Input.sample.html

```diff
@@ -261,4 +289,226 @@ <h3>Input with Label</h3>
+			valueHelpIcon.addEventListener("click", showDialog);
+			searchIcon.addEventListener("click", loadList);
+			searchInput.addEventListener("change", loadList);
+			cancelButton.addEventListener("click", closeDialog);
+			list.addEventListener("item-click", handleItemClick);
+
+			async function loadSuggestions() {
+				var response = await fetch("../../../assets/data/products.json");
+				var products = await response.json();
+
+				var query = valueHelpInput.value.toLowerCase();
+				var suggestionItems = [];
+
+				if (query) {
+					suggestionItems = products
+						.map(function (product) {
+							return product.Name;
+						})
+						.filter(function (productName, index) {
+							return productName.toLowerCase().indexOf(query) === 0;
+						})
+						.slice(0, 10);
+				}
+
+				[].slice.call(valueHelpInput.children, 1).forEach(function (child) {
+					valueHelpInput.removeChild(child);
+				});
+
+				suggestionItems.forEach(function (item) {
+					var li = document.createElement("ui5-suggestion-item");
+					li.text = item;
+					valueHelpInput.appendChild(li);
+				});
+			}
+
+			function showDialog() {
+				searchInput.value = valueHelpInput.value;
+				loadList();
+
+				if (screen.width <= 768) {
+					dialog.setAttribute("stretch", "");
+				}
+
+				dialog.show();
+			}
+
+			async function loadList() {
+				var response = await fetch("../../../assets/data/products.json");
+				var products = await response.json();
+				var query = searchInput.value.toLowerCase();
+
+				list.innerHTML = "";
+
+				products
+					.filter(function (product) {
+						return product.Name.toLowerCase().indexOf(query) === 0;
+					})
+					.forEach(function (item) {
+						var li = document.createElement("ui5-li");
+						li.innerHTML = item.Name;
+						li.image = item.ProductPicUrl;
+						li.description = item.ProductId;
+
+						list.appendChild(li);
+					});
+			}
+
+			function handleItemClick(event) {
+				var item = event.detail.item;
+				valueHelpInput.setAttribute("value", item.innerHTML);
+				dialog.close();
+			}
+
+			function closeDialog() {
+				dialog.close();
+			}
+		</script>
+	</div>
+	<pre class="prettyprint lang-html"><xmp>
+<ui5-input id="valuehelp-input" placeholder="Enter product" style="width: 100%" show-suggestions>
+	<ui5-icon id="valuehelp-icon" input-icon slot="icon" name="value-help"></ui5-icon>
+</ui5-input>
+
+<ui5-dialog id="dialog" accessibleName="Products Value Help">
+	<div slot="header" class="dialog-header">
+		<h2 class="header-title">Products</h2>
+		<ui5-input id="search-input" placeholder="Search" show-clear-icon>
+			<ui5-icon id="search-icon" input-icon slot="icon" name="search"></ui5-icon>
+		</ui5-input>
+	</div>
+
+	<ui5-list id="items-list"></ui5-list>
+
+	<div slot="footer" class="footer">
+		<ui5-button design="Transparent" id="cancel-button">Cancel</ui5-button>
+	</div>
+</ui5-dialog>
+
+<script>
+	var valueHelpInput = document.getElementById("valuehelp-input");
```

Comment:
if you use camelCase IDs, for example `itemList`, then you will be able to access ii directly, as it's automatically bind to the window object.
```js
itemList.addEventListener("item-click", handleItemClick);
```

And you would not need to declare it
var list = document.getElementById("items-list");

and you will skip all that part, which is not valuable for the readers:
```js
    var dialog = document.getElementById("dialog");
	var list = document.getElementById("items-list");
	var searchInput = document.getElementById("search-input");
	var searchIcon = document.getElementById("search-icon");
	var cancelButton = document.getElementById("cancel-button");
```

In addition, you can skip classes to squeeze the mark-up a bit more